pennylane/transforms/core/transform_program.py

```diff
@@ -130,6 +130,12 @@ def __getitem__(self, idx):
     def __bool__(self):
         return bool(self._transform_program)
 
+    def __add__(self, other):
+        program = TransformProgram(self._transform_program)
```

Comment:
do we think `TransformProgram` will remain completely stateless aside from `_transform_program`? It might be worthwhile to add a `__copy__` dunder too, but not super important today.

Reply:
For the moment it is, we can come back to it later